runtime/src/nonce_keyed_account.rs

```diff
@@ -12,6 +13,19 @@ use {
     std::collections::HashSet,
 };
 
+const NONCE_HASH_PREFIX: &[u8] = "NONCE".as_bytes();
+
+fn get_nonce_from_context(invoke_context: &InvokeContext) -> Hash {
```

Comment:
For type safety can we introduce a new `struct Nonce(Hash);`  type and return it here and in `nonce::Data`?

Reply:
done